### PR TITLE
Add folder reading support into SqlParquetReaderWrapper

### PR DESCRIPTION
**Pull request by guillaumer63**

### CODE REVIEW

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java [hunk 1]
```diff
@@ -17,47 +17,102 @@
...
+  }
+
+  private Iterator<ParquetReader<Record>> createReadersFromFolderAndInitSchema(FileSystem fileSystem, Path filePath,
+                                                                               Configuration configuration) throws IOException {
+    List<ParquetReader<Record>> parquetReaderList = new ArrayList<>();
+    for (FileStatus currentFileStatus : fileSystem.listStatus(filePath, HiddenFileFilter.INSTANCE)) {
+      InputFile currentInputFile = HadoopInputFile.fromStatus(currentFileStatus, configuration);
+
+      if (schema == null) {
+        this.initFileSchema(currentInputFile);
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:65]: Parquet support schema evolution, it means that for a parquet file, partitions can have different schemas (but compatibles).  
So by setting the schema according to the first readed file, we can have side effects. A common approach could be "schema merging", create a single schema based on schema of each partition. It's expensive becatse it required to open/close each partition file.   
We can manage this case in other PR, schema merging is not mandaory to read partitioned files.

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java [hunk 2]
```diff
@@ -17,47 +17,102 @@
...
-  private final MessageType schema;
+  private ParquetReader<Record> currentParquetReader;
+  private final Iterator<ParquetReader<Record>> parquetReaderIterator;
+  private MessageType schema;
 
-  public SQLParquetReaderWrapper(String filePath) throws IOException {
-    InputFile inputFile = HadoopInputFile.fromPath(new Path(filePath), new Configuration());
+  public SQLParquetReaderWrapper(String stringFilePath) throws IOException {
+    Path filePath = new Path(stringFilePath);
+    Configuration configuration = new Configuration();
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:47]: To work in different contexts, clients should be able to provide their own hadoop conf. 
I suggest adding a constructor with Configuration in parameter.

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java (hunk 1 above)
Comment (guillaumer63) [reply src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:65]: Ok, was not aware of this. I will create an Issue with that enhancement. Thanks for the tip!

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java (hunk 2 above)
Comment (guillaumer63) [reply src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:47]: True. Change incoming!

**Pull request merged**